.NET/Microsoft.Recognizers.Text.DataDrivenTests/TestHelpers.cs

```diff
@@ -0,0 +1,313 @@
+            switch (parserName)
+            {
+                case "Date":
+                    return new BaseDateParser(new SpanishDateParserConfiguration(commonConfiguration));
+                case "Time":
+                    return new BaseTimeParser(new SpanishTimeParserConfiguration(commonConfiguration));
+                case "DatePeriod":
+                    return new BaseDatePeriodParser(new SpanishDatePeriodParserConfiguration(commonConfiguration));
+                case "TimePeriod":
+                    return new BaseTimePeriodParser(new SpanishTimePeriodParserConfiguration(commonConfiguration));
+                case "DateTime":
+                    return new BaseDateTimeParser(new SpanishDateTimeParserConfiguration(commonConfiguration));
+                case "DateTimePeriod":
+                    return new BaseDateTimePeriodParser(new SpanishDateTimePeriodParserConfiguration(commonConfiguration));
+                case "Duration":
+                    return new BaseDurationParser(new SpanishDurationParserConfiguration(commonConfiguration));
+                case "Holiday":
+                    return new BaseHolidayParser(new SpanishHolidayParserConfiguration());
+                case "Set":
+                    return new BaseSetParser(new SpanishSetParserConfiguration(commonConfiguration));
+                case "Merged":
+                    return new BaseMergedParser(new SpanishMergedParserConfiguration());
+            }
+            return null;
+        }
+
+        private static IModel GetCustomModelFor(string language)
+        {
+            switch (language)
+            {
+                case Culture.Chinese:
+                    return new NumberModel(
+                        AgnosticNumberParserFactory.GetParser(AgnosticNumberParserType.Number, new ChineseNumberParserConfiguration()),
+                        new NumberExtractor(ChineseNumberMode.ExtractAll));
+            }
+            return null;
+        }
+    }
+
+    public static class TestModelExtensions
+    {
+        public static bool IsNotSupported(this TestModel testSpec)
+        {
+            return testSpec.NotSupported.HasFlag(Platform.dotNet);
+        }
+
+        public static bool IsNotSupportedByDesign(this TestModel testSpec)
+        {
+            return testSpec.NotSupportedByDesign.HasFlag(Platform.dotNet);
+        }
+
+        public static DateObject GetReferenceDateTime(this TestModel testSpec)
+        {
+            object dateTimeObject;
+            if (testSpec.Context.TryGetValue("ReferenceDateTime", out dateTimeObject))
+            {
+                return (DateObject)dateTimeObject;
+            }
+            return DateObject.Now;
+        }
+    }
+
+    public static class TestUtils
+    {
+        public static string GetCulture(string source)
+        {
+            var langStr = source.Substring(source.LastIndexOf('_') + 1);
+            switch (langStr)
+            {
+                case "Chs":
```

Comment:
Same as above for "langStr".

Reply:
Updated to use Culture's cultureName instead of hardcoded CHS, SPA, ENG. The Specs folder structure was updated accordingly